src/recipe/Baked.Recipe.Service.Application/Communication/Http/Client.cs

```diff
@@ -8,7 +8,9 @@ public class Client<T>(
     HttpClientFactory<T> _clientFactory
 ) : IClient<T>
 {
-    public async Task<Response> Send(Request request)
+    public async Task<Response> Send(Request request,
+        bool ensureSuccessStatusCode = true
```

Comment:
we don't have to borrow the same naming from http client implementation of .net. we could use a `throwWhenFail` sort of naming